protocol_rfcs/type-widening.md

```diff
@@ -121,16 +121,19 @@ When Type Widening is enabled (when the table property `delta.enableTypeWidening
 - Writers must record type change information in the `metadata` of the nearest ancestor [StructField](#struct-field). See [Type Change Metadata](#type-change-metadata).
 
 When Type Widening is supported (when the `writerFeatures` field of a table's `protocol` action contains `enableTypeWidening`), then:
+- Writers must preserve the `delta.typeChanges` field in the metadata fields in the schema when a schema is updated.
+- Writers can remove an element from a `delta.typeChanges` field in the metadata fields in the schema when all active `add` actions in the latest version of the table have a `defaultRowCommitVersion` value greater or equal to the `tableVersion` value of that `delta.typeChanges` element.
```

Comment:
Since `defaultRowCommitVersion` can be NULL (for files added before this feature was enabled), we should probably explicitly state that all of them have to be non-NULL.
```suggestion
- Writers can remove an element from a `delta.typeChanges` field in the metadata fields in the schema when all active `add` actions in the latest version of the table have a `defaultRowCommitVersion` value that is not NULL and that is greater or equal to the `tableVersion` value of that `delta.typeChanges` element.
```